pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -555,7 +578,17 @@ func (gce *GCECloud) EnsureLoadBalancer(name, region string, requestedIP net.IP,
 	// Deal with the firewall next. The reason we do this here rather than last
 	// is because the forwarding rule is used as the indicator that the load
 	// balancer is fully created - it's what getLoadBalancer checks for.
-	firewallExists, firewallNeedsUpdate, err := gce.firewallNeedsUpdate(name, region, ipAddress, ports)
+	// Check if user specified the allow source range
+	sourceRanges := []string{defaultLBSourceRange}
+	val, ok := annotations[gceLBAllowSourceRange]
+	if ok {
+		if err := gceAnnotationValidation[gceLBAllowSourceRange](val); err != nil {
+			return nil, err
```

Comment:
what does the servicecontroller do with this error? retry till the annotation is correct? do we send an event or something